dashboard/app/controllers/programming_environments_controller.rb

```diff
@@ -14,12 +14,23 @@ def update
       render :not_found
       return
     end
-    programming_environment.assign_attributes(programming_environment_params)
+    programming_environment.assign_attributes(programming_environment_params.except(:categories))
     begin
-      if programming_environment.changed?
```

Comment:
it looks like you are no longer checking if programming environment changes before save! and write_serialization. Is that on purpose?

Reply:
Yeah because I think `changed?` wouldn't cover the case where the categories have changed. I can add back that check for `save!` (that was more of an unintended side effect) but I think `write_serialization` needs to be done either way

Reply:
I don't think you need to. Just was wondering. Probably safer to save always anyway